### PR TITLE
[Rule Tuning] Ensure host information is in endpoint rule queries

### LINKED ISSUE #2286: [Rule Tuning] Add the OS to the query of all endpoint rules
Adding the host information to the rule logic ensures there aren't any accidental collisions with other OS's on the same datasource.

- [ ] Every endpoint rule should use the [field](https://www.elastic.co/guide/en/ecs/current/ecs-os.html#field-os-type) `host.os.type`
- [ ] I also propose adding unit tests for any rule under `rules/(macos|linux|windows)` to enforce it.
- [ ] Linux should have target distributions too (need to verify the most consistent field)
   - [ ] also should unit test this
- [ ] need to verify consistency of all these fields across data sources
   - [ ] endpoint
   - [ ] winlogbeat
   - [ ] system
   - [ ] auditbeat
   - [ ] others?

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
resolves #2286 

## Summary
Adds unit tests to ensure `host.os.name` is in endpoint rule queries and `host.os.platform` is in Linux queries.

Going to hold off adding `host.os.platform` to linux rules until further discussions

#### Field values accross different endpoint data sources:

<details>
<summary>Endgame (`endgame-*`)</summary>

Tested with: Linux and Windows + Telemetry

[Linux Log Sample](https://gist.github.com/w0rk3r/95cdbfcaeaa864aa45e55045ac470a38)
[Windows Log Sample](https://gist.github.com/w0rk3r/47f3b00a0478f182ccb5afadcbd06b00)

`host.os.type` values: "windows", "linux" (No telemetry for this field, and no MacOS to reproduce)
`host.os.name` values: "Windows", "Linux"
`host.os.platform` values: "windows", "linux", "macos"

</details>

<details>
<summary>Endpoint (`logs-endpoint.events.*`)</summary>

Tested with: Linux and Windows + Telemetry

`host.os.type` values: "windows", "linux", "macos"
`host.os.platform` values: "windows", "linux", "macos", "ubuntu", "centos", "unknown distribution", "debian", "red hat", "opensuse", "suse", "fedora", "oracle linux", "amazon linux", "sles", "Windows"

</details>

<details>
<summary>Auditbeat (`auditbeat.*`)</summary>

Tested with: Linux + Telemetry (No MacOS telemetry)

[Auditbeat Linux Log Sample](https://gist.github.com/w0rk3r/dc9cbca2ecbabb13494e2f9c76e61702) 

`host.os.type` values: "linux"
`host.os.name` values: "Debian GNU/Linux", "Amazon Linux", "CentOS Linux", "Ubuntu", "Red Hat Enterprise Linux", "Sangoma Linux", "Rocky Linux", "Red Hat Enterprise Linux Server", "Kali GNU/Linux", "Amazon Linux AMI", "openSUSE Leap", "Oracle Linux Server", "Raspbian GNU/Linux", "AlmaLinux", "CentOS Stream", "Garuda Linux"
`host.os.platform` values: "debian", "amzn", "centos", "ubuntu", "rhel", "sangoma", "rocky", "kali", "opensuse-leap", "ol", "raspbian", "almalinux", "garuda"

</details>

<details>
<summary>Winlogbeat (`winlogbeat-*`)</summary>

Tested with: Windows and Telemetry

[Windows Winlogbeat Log Sample](https://gist.github.com/w0rk3r/2e92919b4bd0f850b1f91b44d0a8ec9a)

`host.os.type` values: "windows"
`host.os.name` values: "Windows Server 2016 Datacenter Evaluation", "Windows Server 2012 R2 Datacenter", "Windows Server 2012 R2 Standard", "Windows Server 2019 Standard", "Windows Server 2016 Standard", "Windows Server 2016 Datacenter", "Windows 10 Enterprise", "Windows Server 2019 Datacenter", "Windows Server 2022 Standard", "Windows 10 Pro", "Windows Server 2012 Datacenter", "Windows Server 2008 R2 Enterprise", "Windows 7 Professional", "Windows Server 2008 R2 Standard", "Windows 11 Pro", "Windows Server 2022 Standard Evaluation", "Windows 11 Home Single Language", "Windows Server 2022 Datacenter", "Windows 11 Enterprise Evaluation", "Windows 10 Enterprise Evaluation", "Windows Server 2019 Standard Evaluation", "Windows 11 Home", "Windows 10 Home", "Windows Server 2012 R2 Datacenter Evaluation", "Windows Server 2022 Datacenter Azure Edition", "Windows 10 Enterprise LTSC 2021", "Windows Storage Server 2012 Standard", "Windows Server 2012 R2 Standard Evaluation", "Windows 10 Enterprise 2016 LTSB", "Windows 10 Pro N", "Windows Server 2012 Standard", "Windows 11 Enterprise", "Windows 10 Enterprise LTSC 2019", "Windows 7 Enterprise", "Windows 10 Pro for Workstations", "Windows 7 Home Premium", "Windows Server 2019 Datacenter Evaluation", "Windows 10 Education"
`host.os.platform` values: "windows"

</details>

<details>
<summary>System Integration (`logs-system.*`)</summary>

Tested with: Linux and Windows + Telemetry (No MacOS telem)

`host.os.type` values: "windows", "linux"
`host.os.name` values: "Windows Server 2019 Standard", "Windows Server 2012 R2 Datacenter", "Windows 10 Enterprise", "Windows Server 2016 Standard", "Windows Server 2012 R2 Standard", "Windows 10 Enterprise for Virtual Desktops", "Windows Server 2019 Datacenter", "Windows Server 2016 Datacenter", "Windows 10 Pro", "Windows 11 Home", "Windows Server 2012 Datacenter", "Windows Server 2022 Standard", "Windows 11 Pro", "Windows 11 Enterprise", "Windows Server 2022 Datacenter", "Windows Server 2008 R2 Standard", "Ubuntu", "Windows Server 2022 Datacenter Azure Edition", "Windows Server 2012 Standard", "Debian GNU/Linux", "Windows Server 2022 Datacenter Evaluation", "Windows 10 Enterprise Evaluation", "Windows Server 2008 R2 Datacenter", "Azure Stack HCI", "CentOS Linux", "Windows 10 Pro N", "Windows 10 Home", "Red Hat Enterprise Linux Server", "Amazon Linux", "Windows Small Business Server 2011 Standard", "Windows 10 Pro for Workstations", "Windows Server 2019 Standard Evaluation", "Windows Server 2008 R2 Enterprise", "Windows Server 2022 Standard Evaluation", "Windows Server 2019 Essentials", "CentOS Stream", "Windows Server 2016 Standard Evaluation", "Red Hat Enterprise Linux", "Windows 11 Enterprise Evaluation", "Windows 10 Enterprise LTSC 2019", "Windows 11 Enterprise multi-session", "Windows Storage Server 2016 Standard", "Windows 10 Pro Education", "OceanX Linux", "Windows 11 Home Single Language", "Windows 10 Education", "Windows 10 Home Single Language", "Windows 8.1 Pro with Media Center", "Windows Server 2019 Datacenter Evaluation", "Hyper-V Server 2019", "Windows 8.1 Pro", "Windows 7 Professional", "SLES", "Windows 10 Enterprise N", "Windows 10 Enterprise 2016 LTSB", "Rocky Linux", "Oracle Linux Server", "CentOS", "Kali GNU/Linux", "Zorin OS", "AlmaLinux", "CloudLinux", "TencentOS Server", "Virtuozzo"
`host.os.platform` values: "windows", "ubuntu", "debian", "centos", "rhel", "amzn", "oceanx", "sles", "rocky", "ol", "kali", "zorin", "almalinux", "cloudlinux", "tencentos", "virtuozzo"

</details>

<details>
<summary>Windows Integration (`logs-windows.*`)</summary>

Tested with: Windows + Telemetry

`host.os.type` values: "windows"
`host.os.name` values: "Windows 10 Enterprise", "Windows 10 Pro", "Windows Server 2016 Datacenter", "Windows Server 2012 R2 Standard", "Windows Server 2012 R2 Datacenter", "Windows Server 2022 Datacenter", "Windows Server 2019 Datacenter", "Windows Server 2016 Standard", "Windows Server 2012 Datacenter", "Windows Server 2019 Standard", "Windows 10 Enterprise for Virtual Desktops", "Windows Server 2022 Datacenter Azure Edition", "Windows 10 IoT Enterprise", "Windows 11 Enterprise", "Windows 10 Enterprise LTSC 2019", "Windows Server 2019 Standard Evaluation", "Windows 11 Pro", "Hyper-V Server 2019", "Windows 8.1 Pro with Media Center", "Windows 11 Home Single Language", "Windows 10 Enterprise Evaluation", "Windows Server 2022 Standard", "Windows Server 2012 Standard", "Windows 10 Home", "Windows 11 Enterprise multi-session", "Windows Server 2022 Standard Evaluation", "Windows 8.1 Pro", "Windows 10 Enterprise Evaluation", "Windows 10 Education", "Windows 10 Pro for Workstations", "Windows 11 Home", "Windows Server 2008 R2 Enterprise", "Azure Stack HCI", "Windows 11 Pro for Workstations", "Windows Server 2008 R2 Standard", "Windows 10 Pro Education", "Windows 7 Professional", "Windows Server 2008 R2 Datacenter", "Windows 7 Enterprise", "Windows Server 2016 Standard Evaluation", "Windows 10 Enterprise N", "Windows 10 Pro N", "Windows 7 Ultimate", "Windows Server 2019 Datacenter Evaluation", "Hyper-V Server 2012 R2", "Windows 10 Enterprise 2016 LTSB"
`host.os.platform` values: "windows"

</details>

## Testing

To test, simply remove the `host.os.type` from any endpoint rule and run `make test`

### CODE REVIEW

File: rules/macos/persistence_login_logout_hooks_defaults.toml [hunk 1]
```diff
@@ -33,7 +33,7 @@ timestamp_override = "event.ingested"
 type = "eql"
 
 query = '''
-process where event.type == "start" and
+process where host.os.name == "macos" and event.type == "start" and
```
Comment (brokensound77) [inline rules/macos/persistence_login_logout_hooks_defaults.toml:36]: This failure seems to be a bug in the parsing logic.

error:
```
FAILED tests/test_all_rules.py::TestEndpointQuery::test_os_and_platform_in_query - AssertionError: 'host.os.name' not found in [] : 5d0265bf-dea9-41a9-92ad-48a8dcd05080 - Persistence via Login or Logout Hook -> missing required field for endpoint rule
```

No fields were parsed: `[]`

```
rule.name
'Persistence via Login or Logout Hook'

rule.id
'5d0265bf-dea9-41a9-92ad-48a8dcd05080'

rule.contents.data.query
'process where host.os.name == "macos" and event.type == "start" and\n process.name == "defaults" and process.args == "write" and process.args in ("LoginHook", "LogoutHook") and\n not process.args :\n       (\n         "Support/JAMF/ManagementFrameworkScripts/logouthook.sh",\n         "Support/JAMF/ManagementFrameworkScripts/loginhook.sh",\n         "/Library/Application Support/JAMF/ManagementFrameworkScripts/logouthook.sh",\n         "/Library/Application Support/JAMF/ManagementFrameworkScripts/loginhook.sh"\n       )\n'

rule.contents.data.ast
PipedQuery(first=EventQuery(event_type='process', query=Boolean(value=False)), pipes=[])

str(rule.contents.data.ast)
'process where false'
```

The original logic is parsed simply into `process where false`

File: tests/test_all_rules.py [hunk 2]
```diff
@@ -814,3 +815,27 @@ def test_note_guide(self):
...
+                # skip cross-platform for now
+                continue
+
+            ast = rule.contents.data.ast
+            fields = [str(f) for f in ast if isinstance(f, (kql.ast.Field, eql.ast.Field))]
+
+            err_msg = f'{self.rule_str(rule)} missing required field for endpoint rule'
+            self.assertIn('host.os.name', fields, err_msg)
+
+            # going to bypass this for now
```
Comment (brokensound77) [inline tests/test_all_rules.py:838]: I think for now we can skip the platform test and enable in a later PR

**Comment by brokensound77** [comment]
Pending @w0rk3r datasources review and the weird parsing error, this is ready for review

File: rules/macos/persistence_login_logout_hooks_defaults.toml (hunk 1 above)
Comment (brokensound77) [reply rules/macos/persistence_login_logout_hooks_defaults.toml:36]: looks like certain line breaks are legal

image

image

**Comment by brokensound77** [comment]
@w0rk3r can you also add `host.os.type` to your review. Endpoint agent has `host.os.name` with title casing (`Linux`) and so we would need to use `host.os.name.caseless` which is less ideal since it may not be consistent. It does have `host.os.type` as lowercased though, so it may be the better option.

**Comment by w0rk3r** [comment]
@brokensound77 I think we should modify the index `logs-system.auth-*` to point to `logs-system.*` which is the standard on the windows side

**Comment by brokensound77** [comment]
> @brokensound77 I think we should modify the index `logs-system.auth-*` to point to `logs-system.*` which is the standard on the windows side

That makes sense to me, for those targeting the `auth` dataset specifically. You can do that in a separate PR though.

File: rules/macos/persistence_login_logout_hooks_defaults.toml [hunk 3]
```diff
@@ -33,8 +33,8 @@ timestamp_override = "event.ingested"
 type = "eql"
 
 query = '''
-process where event.type == "start" and
- process.name == "defaults" and process.args == "write" and process.args in ("LoginHook", "LogoutHook") and
+process where host.os.type == "macos" and event.type == "start" and
+ process.name == "defaults" and process.args == "write" and process.args : ("LoginHook", "LogoutHook") and
```
Comment (brokensound77) [inline rules/macos/persistence_login_logout_hooks_defaults.toml:37]: FYSA: I changed the `in` to `:` to circumvent the parsing bug. Should have no negative impact on the logic (and can be changed back once resolved).

REF: #2611

cc: @DefSecSentinel

**Comment by brokensound77** [comment]
Thanks for adding the sample logs @w0rk3r!

File: tests/test_all_rules.py [hunk 4]
```diff
@@ -814,3 +815,27 @@ def test_note_guide(self):
...
                     self.fail(f'{self.rule_str(rule)} Investigation guides using the Osquery Markdown must contain '
                               f'the following note:\n{osquery_note_pattern}')
+
+
+class TestEndpointQuery(BaseRuleTest):
+    """Test endpoint-specific rules."""
+
+    def test_os_and_platform_in_query(self):
+        """Test that all endpoint rules have an os defined and linux includes platform."""
+        for rule in self.production_rules:
```
Comment (Mikaayenson) [inline tests/test_all_rules.py:825]: If we only iterate over production rules, we'll miss out on rules with `maturity = "development"`.  Since we're already tuning them e.g. `defense_evasion_msbuild_beacon_sequence.toml`, we probably just expand to all rules.

File: tests/test_all_rules.py [hunk 5]
```diff
@@ -814,3 +815,27 @@ def test_note_guide(self):
...
+            if not rule.contents.data.get('language') in ('eql', 'kuery'):
+                continue
+            if rule.path.parent.name not in ('windows', 'macos', 'linux'):
+                # skip cross-platform for now
+                continue
```
Comment (Mikaayenson) [inline tests/test_all_rules.py:830]: Is there any reason why we can't add the new field to the cross-platform rules?

File: tests/test_all_rules.py [hunk 6]
```diff
@@ -814,3 +815,27 @@ def test_note_guide(self):
...
+            ast = rule.contents.data.ast
+            fields = [str(f) for f in ast if isinstance(f, (kql.ast.Field, eql.ast.Field))]
+
+            err_msg = f'{self.rule_str(rule)} missing required field for endpoint rule'
+            self.assertIn('host.os.type', fields, err_msg)
+
+            # going to bypass this for now
+            # if rule.path.parent.name == 'linux':
+            #     err_msg = f'{self.rule_str(rule)} missing required field for linux endpoint rule'
+            #     self.assertIn('host.os.platform', fields, err_msg)
```
Comment (Mikaayenson) [inline tests/test_all_rules.py:841]: One item of the checklist in the issue for this PR is:

` Linux should have target distributions too (need to verify the most consistent field)` and  `also should unit test this`. Is this task item relevant to this section of code?

**Review by Mikaayenson — COMMENTED** [review]
- I modified a rule and ran pytest which successfully caught the missing field.

<details>
<summary> Pytest Output </summary>

```bash

    (detection_dev) ➜  detection-rules git:(2286-add-os-and-platform) ✗ pytest
======================================================================================================================================================================================================================================================================= test session starts =======================================================================================================================================================================================================================================================================
platform darwin -- Python 3.10.9, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/stryker/workspace/Community/brokensound77/detection-rules, configfile: pyproject.toml
plugins: typeguard-2.13.3, profiling-1.7.0
collected 118 items

tests/test_all_rules.py ..................s..........F                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [ 25%]
tests/test_gh_workflows.py .                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 26%]
tests/test_mappings.py ...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  [ 28%]
tests/test_packages.py .....                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 33%]
tests/test_schemas.py ..........                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 41%]
tests/test_toml_formatter.py ...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [ 44%]
tests/test_utils.py ....                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    [ 47%]
tests/test_version_locking.py .                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 48%]
tests/kuery/test_dsl.py .......                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 54%]
tests/kuery/test_eql2kql.py .........                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 61%]
tests/kuery/test_evaluator.py .............                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ 72%]
tests/kuery/test_kql2eql.py ..........                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [ 81%]
tests/kuery/test_lint.py ............                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [ 91%]
tests/kuery/test_parser.py ..........                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [100%]

============================================================================================================================================================================================================================================================================ FAILURES =============================================================================================================================================================================================================================================================================
_________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________ TestEndpointQuery.test_os_and_platform_in_query _________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________

self = <tests.test_all_rules.TestEndpointQuery testMethod=test_os_and_platform_in_query>

    def test_os_and_platform_in_query(self):
        """Test that all endpoint rules have an os defined and linux includes platform."""
        for rule in self.production_rules:
            if not rule.contents.data.get('language') in ('eql', 'kuery'):
                continue
            if rule.path.parent.name not in ('windows', 'macos', 'linux'):
                # skip cross-platform for now
                continue

            ast = rule.contents.data.ast
            fields = [str(f) for f in ast if isinstance(f, (kql.ast.Field, eql.ast.Field))]

            err_msg = f'{self.rule_str(rule)} missing required field for endpoint rule'
>           self.assertIn('host.os.type', fields, err_msg)
E           AssertionError: 'host.os.type' not found in ['event.action', 'process.name', 'process.Ext.effective_parent.name', 'process.Ext.effective_parent.executable'] : 1dee0500-4aeb-44ca-b24b-4a285d7b6ba1 - Suspicious Inter-Process Communication via Outlook -> missing required field for endpoint rule

tests/test_all_rules.py:836: AssertionError
===================================================================================================================================================================================================================================================================== short test summary info =====================================================================================================================================================================================================================================================================
FAILED tests/test_all_rules.py::TestEndpointQuery::test_os_and_platform_in_query - AssertionError: 'host.os.type' not found in ['event.action', 'process.name', 'process.Ext.effective_parent.name', 'process.Ext.effective_parent.executable'] : 1dee0500-4aeb-44ca-b24b-4a285d7b6ba1 - Suspicious Inter-Process Communication via Outlook -> missing required field for endpoint rule
============================================================================================================================================================================================================================================================ 1 failed, 116 passed, 1 skipped in 43.96s ============================================================================================================================================================================================================================================================
(detection_dev) ➜  detection-rules git:(2286-add-os-and-platform) ✗

```
</details>

- I saw a task item in the original issue to `verify consistency of all these fields across data sources`, but I don't see mention of that in this PR. Is there more to do?
- Still need to discuss adding the `host.os.platform` field to linux rules but here is some initial feedback.

File: tests/test_all_rules.py (hunk 4 above)
Comment (brokensound77) [reply tests/test_all_rules.py:825]: the whole point of rules in development is to not force compliance while they are in dev

File: tests/test_all_rules.py (hunk 6 above)
Comment (brokensound77) [reply tests/test_all_rules.py:841]: ywp, distribution == platform

**Comment by brokensound77** [comment]
@Mikaayenson 

>I saw a task item in the original issue to verify consistency of all these fields across data sources, but I don't see mention of that in this PR. Is there more to do?

Go to the description of this PR and expand all the individual `details`

image

@w0rk3r did an awesome job analyzing all the unique datasources used in our endpoint rules


>Still need to discuss adding the host.os.platform field to linux rules but here is some initial feedback.

I am going to open a dedicated issue for this to ensure it gets the proper discussion

File: rules/macos/persistence_login_logout_hooks_defaults.toml (hunk 3 above)
Comment (DefSecSentinel) [reply rules/macos/persistence_login_logout_hooks_defaults.toml:37]: 👍

**Review by DefSecSentinel — APPROVED** [review]
LGTM!

File: rules/windows/persistence_adobe_hijack_persistence.toml [hunk 7]
```diff
@@ -39,8 +39,8 @@ Attackers can replace the `RdrCEF.exe` executable with their own to maintain the
...
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services where host.os.type == "windows" and NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path where host.os.type == "windows" and authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```
Comment (w0rk3r) [inline rules/windows/persistence_adobe_hijack_persistence.toml:43]:
```suggestion
      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```

File: rules/windows/persistence_evasion_registry_startup_shell_folder_modified.toml [hunk 8]
```diff
@@ -42,8 +42,8 @@ Techniques used within malware and by adversaries often leverage the Windows reg
...
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services where host.os.type == "windows" and NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path where host.os.type == "windows" and authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```
Comment (w0rk3r) [inline rules/windows/persistence_evasion_registry_startup_shell_folder_modified.toml:46]:
```suggestion
      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```

File: rules/windows/persistence_priv_escalation_via_accessibility_features.toml [hunk 9]
```diff
@@ -48,8 +48,8 @@ This rule looks for the execution of supposed accessibility binaries that don't
...
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services where host.os.type == "windows" and NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path where host.os.type == "windows" and authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```
Comment (w0rk3r) [inline rules/windows/persistence_priv_escalation_via_accessibility_features.toml:52]:
```suggestion
      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```

File: rules/windows/persistence_run_key_and_startup_broad.toml [hunk 10]
```diff
@@ -42,8 +42,8 @@ Adversaries may achieve persistence by referencing a program with a registry run
...
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services where host.os.type == "windows" and NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path where host.os.type == "windows" and authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```
Comment (w0rk3r) [inline rules/windows/persistence_run_key_and_startup_broad.toml:46]:
```suggestion
      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE '%LocalSystem' OR user_account LIKE '%LocalService' OR user_account LIKE '%NetworkService' OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != 'trusted'","label":"Osquery - Retrieve Service Unsigned Executables with Virustotal Link"}}
```

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Left some suggestions around unintended replacements, duplicated logic, and missing bumps on updated_date

**Review by Mikaayenson — APPROVED** [review]

File: rules/windows/credential_access_lsass_memdump_handle_access.toml [hunk 11]
```diff
@@ -108,7 +108,7 @@ timestamp_override = "event.ingested"
 type = "eql"
```
Comment (w0rk3r) [inline rules/windows/credential_access_lsass_memdump_handle_access.toml:108]: Missing bump on updated_date

File: rules/windows/discovery_privileged_localgroup_membership.toml [hunk 12]
```diff
@@ -91,7 +91,7 @@ timestamp_override = "event.ingested"
 type = "eql"
```
Comment (w0rk3r) [inline rules/windows/discovery_privileged_localgroup_membership.toml:91]: Missing bump on updated_date

File: rules/windows/execution_from_unusual_path_cmdline.toml [hunk 13]
```diff
@@ -89,7 +89,7 @@ timestamp_override = "event.ingested"
 type = "eql"
```
Comment (w0rk3r) [inline rules/windows/execution_from_unusual_path_cmdline.toml:89]: Missing bump on updated_date

File: rules/linux/defense_evasion_attempt_to_disable_iptables_or_firewall.toml [hunk 14]
```diff
@@ -24,7 +24,7 @@ tags = ["Elastic", "Host", "Linux", "Threat Detection", "Defense Evasion"]
 timestamp_override = "event.ingested"
```
Comment (w0rk3r) [inline rules/linux/defense_evasion_attempt_to_disable_iptables_or_firewall.toml:24]: Missing bump on updated_date

**Review by w0rk3r — APPROVED** [review]
Just some bumps on updated_date and I think we are ready to go

**Comment by brokensound77** [comment]
> Just some bumps on updated_date and I think we are ready to go

Thanks! Just verified all the dates in them reflect 22 Feb 23, which was the original bump date in this PR, so should be g2g
